### ISSUE TITLE
Komodo crashes 5-10 seconds after opening any file

### ISSUE THREAD

**Issue by AvalonNZ** [Komodo/KomodoEdit#3404]
### Short Summary

Windows updated itself overnight and today Komodo started crashing.

Komodo crashes about 5-10 seconds after I open a file (or Komodo open files that were opened in a previous session). 

Note: I've noticed that the Symbol Browser regularly hangs too (intermittently) so maybe it's failing to parse the PHP file?

I get the Windows Dialog box: "ActiveState Komodo has stopped working". It then kills the program regardless of whether I click the button or the "X" in the top right corner.

### Steps to Reproduce

Start Komodo 11 

Open any file (works if I let Komodo open files that were open in the previous session or if I decline to open a previous session and open a file from a clean slate). 

Wait 5-10 seconds and Komodo will crash.

It happens every time I run Komodo.

### Expected results

Open file and start work.

### Actual results

Komodo crashes just after file opens. No work done today.

### Platform Information

Komodo IDE, version 11.0.1, build 90797, platform win32-x86.
Built on Fri Oct 27 14:40:19 2017.
(presumably latest stable build - Komodo reports no updates available)

I'm running the latest version of Windows 10 Pro 64-bit (the Creative Update - v1709, OS build 16299.98) as at December 12, 2017

I have an Intel i7 CPU running standard at 2.79 GHz with 8GB RAM on board, and an NVIDIA GForce 760 graphics card. 

### Additional Information

 
pystderr.log was empty.

**Comment by AvalonNZ** [comment]
Also, I use GIT via SourceTree. I don't use the Komodo GIT integration directly, but I notice that it's tracking the files as I change them.

**Comment by mitchell-as** [comment]
Hi, one issue per ticket please.

If you restart Komodo in Safe Mode does it still crash regularly? (Help > Troubleshooting > Restart in Safe Mode) If so, try re-installing Komodo and then try again with your default profile (run Komodo as normal), and then in Safe Mode.

**Comment by AvalonNZ** [comment]
There is only one issue here. The rest (Git, Symbol browser, etc) is just background information that your people have asked for in other tickets regarding Komodo crashes. I'm just trying to be proactive.

Yes, it still crashes every time in safe mode. 

I'll try reinstalling and get back to you.

**Comment by AvalonNZ** [comment]
There is only one issue here. The rest (Git, Symbol browser, etc) is just background information that your people have asked for in other tickets regarding Komodo crashes. I'm just trying to be proactive.

Yes, it still crashes every time in safe mode. 

I'll try reinstalling and get back to you.

**Comment by AvalonNZ** [comment]
Uninstalled Komodo 11
Reinstalled Komodo 11
Ran Komodo - crashed just seconds after opening a file without even giving me the windows message.
Ran Komodo again. This time it crashed 30 seconds after opening two files (with the usual windows message)
Rebooted the PC.
Ran Komodo again and got it to restart in safe mode. I accepted all defaults. 
[pystderr.log](https://github.com/Komodo/KomodoEdit/files/1549572/pystderr.log)
[codeintel3.log](https://github.com/Komodo/KomodoEdit/files/1549574/codeintel3.log)
[last-update.log](https://github.com/Komodo/KomodoEdit/files/1549575/last-update.log)



I left it alone for 2 minutes, then opened a file. It crashed 4 seconds later.
Note: all of these files are on my local disk. There are no remote connections defined or in use.

pystdout.log is empty.

**Comment by mitchell-as** [comment]
Hi, are you getting a crash report dialog when Komodo crashes? I see in your logs you are opening files on a `G:\` drive. Is that a remotely mapped drive? If you open files on your `C:\` drive do you still get a crash?

**Comment by AvalonNZ** [comment]
As previously mentioned in this ticket:
**I get the Windows Dialog box: "ActiveState Komodo has stopped working"**. It then kills the program regardless of whether I click the button or the "X" in the top right corner.

This is the only notification I get. There's no "crash report dialog" unless you're talking about the Windows dialog I mentioned several times. 

Also as mentioned in the my last post:
Note: all of these files are on my local disk. There are no remote connections defined or in use.

G: is one of five local disks on the PC under my desk. It holds all my source code and graphic files. 
**I still get a crash if I open files on the C: drive**. I opened the pystderr.log file from the ActiveState directory, deep in my c:\user directory, and the same thing happened: It crashed to the "ActiveState Komodo has stopped working" dialog box after about 5 seconds. 

This is a simple home office setup. One PC with all 5 hard-drives held inside the machine. Internet access is via CAT5 to the router. Nothing but a printer attached. There are no NAS devices and I handle all connections to my servers via PuTTY terminals and WinSCP. **Komodo only has access to local files**. GIT is handled locally by SourceTree and Database connections are handled by PuTTY too so Komodo is not aware of any remote connections (although it can see my local GIT, Python, and other language installations).

**Comment by AvalonNZ** [comment]
Further info: The problem exists on Komodo 10 and Komodo 9 too. 

I actually got Komodo 9 to open the last set of files it had open, and then got it to create a new project before I tried opening a file from the Files list (which triggered the usual crash).

**Comment by AvalonNZ** [comment]
In case it helps pin things down, I reinstalled Komodo EDIT 7 and that seems to work fine. I've created two new projects and swapped between them. Most importantly I've opened several files by clicking on a filename in the left sidebar and the program didn't crash.

**Comment by mitchell-as** [comment]
Thanks for the extra information. I did see your original comment on the Windows crash notification. Komodo's crash reporter is separate from Windows'. Komodo's allows you to send the report to us for further evaluation. The fact that you're not getting that is odd.

Sorry I forgot your mentioning of exclusively local file use. I got distracted when reading your log file. (I often map non-C drive files to remote locations.)

The last suggestion I have for you is to temporarily rename or move your Komodo profile directory. You can find this directory here: https://community.activestate.com/faq/komodo-file-locations. Go ahead and rename or move the entire "Komodo" and "Komodo Edit" directories (not just a particular version) in order to ensure you are starting from scratch. I have a suspicion this will not change anything since you still get crashes in Safe Mode, but it doesn't hurt to try...

@Naatan can you think of any reason why Komodo 7 would not cause crashes when opening files as opposed to 9, 10, and 11? I'm beginning to think the Windows update just completely hosed something.

**Comment by AvalonNZ** [comment]
@Naatan My apologies for being imprecise. Komodo EDIT 7 (the last version I used before upgrading) seems to work fine. Komodo IDE 9, 10, and 11 are crashing. My intention was to illustrate that Komodo EDIT seems fine while Komodo IDE has a problem (on the off-chance that there are obvious differences in the underlying architecture that might help you pinpoint the issue). 

I agree that the Windows Update is the likely culprit due to the timing but assumed that this might be affecting a wider group as a result. Maybe it's just me. I've just tried my MS Surface (laptop) which has Komodo IDE 10 installed and that's working ok despite having also received the same MS Update (but it's a significantly different kind of machine so who knows).

Firefox is still working fine on my PC (although that's had a major upgrade recently too, and the version Komodo uses internally is presumably some way behind the public version of Firefox now).  

Assuming the Windows Update hosed something, I'd have hoped that uninstalling and reinstalling Komodo IDE 11 would have fixed the problem. 

I'll try the change to the profile directory and get back to you.

**Comment by AvalonNZ** [comment]
Before I change any directory names, here's a screenshot of the c:\users\Simon\AppData\local directory structure. 

a) which directory do you want me to rename?
b) Is there anything here that raises a concern?
![c-user-simon-appdata-local-activestate](https://user-images.githubusercontent.com/1366229/33966086-2a8e1680-e0c3-11e7-8c03-a4d4dbb0c949.png)

**Comment by AvalonNZ** [comment]
Note: none of the directories in the screenshot are actually selected, which is why the files panel is blank.

**Comment by mitchell-as** [comment]
The "KomodoIDE" and "KomodoEdit" folders (no space in-between words).

**Comment by AvalonNZ** [comment]
Ok. Having two sets of Komodo directories looked weird to me too but presumably Komodo put them there so the question is what do we do about it?

**Comment by mitchell-as** [comment]
I asked you to either temporarily rename or move them. Then start Komodo again and see if you still get crashes (all of your settings will be gone until you restore the original folders). We are trying to determine if something in your profile is causing trouble. Like I said, Safe Mode should rule this out, but I want to double-check.

**Comment by AvalonNZ** [comment]
Tried to move KomodoIDE to a temp directory and was told it's in use by another program, I'm going to reboot and try again.

**Comment by AvalonNZ** [comment]
Rebooted.
Moved the KomodoIDE directory to temp folder.
Started Komodo IDE 11
Komodo opened in safe mode.
I accepted all the defaults without modification by hitting "Next" on each screen.
From the left sidebar, I opened a directory on a drive I haven't accessed since this problem started.
Double clicked a filename in the left sidebar.
File opened ok. 
Started counting to myself.
After a count of ten I opened a second file in the same manner.
I got to a count of 22 when Komodo IDE 11 crashed and the Windows dialog popped up saying "ActiveState Komodo has stopped working".

**Comment by AvalonNZ** [comment]
And it successfully recreated the KomodoIDE directory that I'd moved, and it now contains only one subdirectory for "11.0".

**Comment by mitchell-as** [comment]
I'm out of ideas. The fact that it is working on one of your machines, but not another leads me to think it's a machine-specific problem. @Naatan, any ideas?

**Comment by Naatan** [comment]
I think what might be happening is Komodo is scanning files in your opened directory and running into some type of issue. Could you try pointing your Places widget at another directory, something with less files in it?

Press the Cog icon in the left pane and change the directory. Given Komodo crashes soon after you start it you'll have to be fast about this :\ Try changing the directory and then closing Komodo and reopening it, to ensure your preference is saved and to ensure the crash doesn't still happen and wipes your pref change.

**Comment by AvalonNZ** [comment]
@Naatan Komodo IDE 11 crashes only after a file is opened. So if I start Komodo without restoring the workspace (ie: files and projects) from the last session (via the nice Workspace Restore dialog box that I always get when restarting Komodo IDE after a crash), I can change directories and create projects as long as I like without causing a crash. I've let it sit there doing nothing much for ten minutes with no files open and it's perfectly stable. 

It's only when I open a file (and after the file is displayed in the editing window) that the countdown to the crash begins. I can sometimes open several files in the editor before it crashes, but my experience suggests that the crash is triggered by the first "open file" action, and that I can get between 2-22 seconds after opening that first file before the crash occurs. During that time I can start editing the file or open more files or anything else as the UI has not locked up during that time. 

FYI, I've been a programmer for 35 years, the first 20 of them on various Dos and Windows systems, in a variety of languages including Basic, UCSD Pascal, Quick Basic, MS C++. Dbase, Clipper, and Borland Delphi (for most of the second decade). I'm used to debugging this stuff.

The program always crashes regardless of the number of files I have opened so my guess is that it's the act of opening the first file that causes the crash. 

The UI doesn't lock up prior to the crash, and I can continue to work right up to the crash so my first instinct is that the problem is a background task that's running after the first (existing) file is opened. 

However it could also be related to the clicking of the filename in the left sidebar. To test this theory I tried a few more things.

For a bit of variety I tried creating a PHP file "from Template" and it's fine with that. 
I even typed a few lines of nonsense code into the editing pane (see image 5) and there were no problems. Code Insight ran fine too.

After 4 minutes (using the stopwatch on my phone) I saved the file to disk. Komodo stayed stable. 

After another three minutes I closed the file and a minute later I reopened it from the Files | Recently Closed Tabs list.

To my suprprise it didn't crash. I left it for four minutes and then closed it again.

This time I reopened it from the Files | Recent Files list. That worked fine too. I closed it after 3 minutes.

I used the File | Open | File dialog to reopen it and once again it was fine.  I gave it 90 seconds and then opened a small XML file from the same temp directory. 

For about four minutes it appeared to be stable but as I was going to close the original file I noticed that Komodo was not responding. I don't know exactly when this started. Six minutes after I noticed it was not responding I got the usual crash dialog  box and it closed itself, but I'm not convinced it was related to the main problem because Firefox also became unstable on my PC at about the same time and I had to shut it down too. 

I restarted Firefox and then Komodo IDE. For a change I allowed it to restore the workspace and the nonsense.php file came back. The little xml file that I last opened did not reopen. 

After two minutes I closed the php file using the X on the tab.

I then double clicked the nonsemse.php filename in the left sidebar (which usually causes the crash), The file opened and all was well. I gave it four minutes but nothing bad happened. I edited the file, saved it, and still no crash. 

I closed the little php file and opened an HTML file from the same directory by double clicking on the filename in the left sidebar. Three seconds later Komodo crashed. I barely had time to reset the stopwatch. 

Maybe we're getting somewhere now.

I reopened Komodo IDE 11 and allowed it to restore the workspace. Nonsense.php reopened and I left it alone for two minutes. Then I made a couple of simple code changes and saved the file. I gave it two minutes, then closed the file. I reopened it again from the left sidebar. Three minutes later, all was well so I opened another (larger) php file from the Files | Recent Files list and I got 6 secodns before it crashed. 

The weird thing about this is that it seems to be fine with the little php file I created today. It doesn't seem to like much else. I hope some of this helps.

**Comment by Naatan** [comment]
Thanks for the detailed report @AvalonNZ, however it isn't clear to me whether you changes your directory scope as I asked? It doesn't really matter whether the crash happened on opening a file or something else, Komodo uses your opened directory as the scope for many different things.

In addition, something you could try is turning off all SCC file monitoring prefs. Head over to Prefs > Source Code Control and disable all of them (there's a checkbox next to their title).

**Comment by AvalonNZ** [comment]
Hi Naatan, you're right. I wasn't clear about that, but yes, I have changed directory numerous times using the method you described, without causing a crash. And each time the program crashes only when I open a file from the new directory. 

FYI, I've installed the security and enhancement update to Komodo IDE 11 today. 

- Launched Komodo IDE 11
- installed the security and enhancement update to Komodo IDE 11 
- restarted Komodo
- the g:\temp directory was open on launch
- I opened the nonsense.php file from the left sidebar
- waited two minutes - no crash
- opened an HTML file from the same directory from the left sidebar
- Komodo immediately became non-responsive.
- 5 seconds later it crashed with the "ActiveState Komodo has stopped working" dialog.
= Todays' enhancement wasn't the answer to the problem. 

- Launched Komodo IDE 11
- declined Workplace Restore
- the g:\temp directory was open on launch
- used the Cog icon | Open Directory to change the directory as requested.
- the directory is now G:\pad3\var\www\kpidemo.avalon.net.nz\htdocs
- note: this directory has NO GIT repository attached to it.
- waited two minutes - no crash
- opened header.php (which is a tiny, 1K file) by double clicking on the filename in the left sidebar
- waited two minutes - no crash
- opened footer.php (which is a tiny, 1K file) by double clicking on the filename in the left sidebar
- waited two minutes - no crash
- opened dp_edit6.php (which is a 14K file) by double clicking on the filename in the left sidebar
- Komodo became non-responsive after 8 seconds and crashed 2 seconds later (with the usual message).

[pystderr_1.log](https://github.com/Komodo/KomodoEdit/files/1564334/pystderr_1.log)

- Launched Komodo IDE 11
- declined Workplace Restore
- the G:\pad3\var\www\kpidemo.avalon.net.nz\htdocs direcotry was open on launch
- went into Preferences | SCC panel
- CVS integration was ticked but grayed out (and couldn't be unticked), and the message said it was disabled because no CSV executable could be found. 
- Perforce integration was ticked but grayed out (and couldn't be unticked), and the message said it was disabled because no p4 executable could be found. 
- SVN integration was UN-ticked (presumably because I explicitly disabled it from the popup dialog box that came up on launch at one point this week) and grayed out (and couldn't be unticked), and the message said it was disabled because no SVN executable could be found. 
- Bazaar integration was ticked but grayed out (and couldn't be unticked), and the message said it was disabled because no bzr executable could be found. 
**- Git integration was ticked and white. I unticked it.**
- Mercurial integration was ticked but grayed out (and couldn't be unticked), and the message said it was disabled because no hg executable could be found. 
- I hit Apply and then OK and to dialog box closed.
- used the Cog icon | Open Directory to change the directory 
- the directory is now G:\Nero\var\www\admin.nzmt.org.nz
- note: this directory DOES have a GIT repository attached to it.
- exposed subdirectory /htdocs in the left sidebar
- opened footer.php (which is a tiny, 1K file) by double clicking on the filename in the left sidebar
- waited three minutes - no crash
- opened filesys_copy_file.php (which is a 3K file) by double clicking on the filename in the left sidebar
- Komodo became non-responsive after 9 seconds and crashed 3 seconds later (with the usual message).

[pystderr_2.log](https://github.com/Komodo/KomodoEdit/files/1564355/pystderr_2.log)

**Comment by AvalonNZ** [comment]
In case Komodo IDE 11 needed to be restarted before the SCC settings took hold, I did the following:

- Launched Komodo IDE 11
- declined Workplace Restore
- the G:\Nero\var\www\admin.nzmt.org.nz directory was open on launch
- Selected the project (cli.admin.nzmt) I was working on when this problem first appeared 
- the directory is now G:\Nero\var\www\admin.nzmt.org.nz\cli directory 
- used the Cog icon to change directory to G:\Nero\var\www\admin.nzmt.org.nz\htdocs
- waited three minutes - no crash
- went into Preferences | SCC panel
- All settings are as I left them. All say they're disabled due to missing executables, except Git, which is unticked, but there's no message saying it's disabled. 
- Cancelled Preferences dialog.
- double clicked ajax_delete_records.php in the left sidebar to open file
- started mousing around in the editing window, scrolling up and down, clicking in the window every 1-2 seconds for the first 25 seconds. I have no real expectation that this is relevant to the problem but it's something new to try.
- waited three minutes - no crash
- opened footer.php (which is a tiny, 1K file) by double clicking on the filename in the left sidebar
- waited two minutes - no crash
- opened class_prescribing_term_index.php (which is a 10K file) by double clicking on the filename in the left sidebar
- waited 1:55 minutes (and was just about to open another file) when Komodo crashed again.

[pystderr_3.log](https://github.com/Komodo/KomodoEdit/files/1564376/pystderr_3.log)

**Comment by Naatan** [comment]
Does the crash always happen on relatively large files? Do you see any other correlations between file types and crashes?

**Comment by AvalonNZ** [comment]
I don't think I've noticed any specific correlations with file types. I've opened mainly php files of varying sizes. However I've also had crashes on HTML files. 

It looks like smaller files (<2k) last longer or don't trigger the crash. I just opened about five php files that were less than half a page of code (7-8 lines , around 1k) and another that was about a page and a half of code (about 2k). Komodo didn't crash. 

After about 20 minutes of this I opened a much bigger file and it crashed within seconds. 

Size may well be an issue.

**Comment by Naatan** [comment]
Could you try disabling all the codeintel features (Preferences > CodeIntel), then restart Komodo and see if the crash still happens?

**Comment by AvalonNZ** [comment]
Ok. I disabled Code Intel and restarted Komodo IDE 11.

![pic_10_preferences_code_intel_after_disabling_everything](https://user-images.githubusercontent.com/1366229/34226221-e4859ba2-e62d-11e7-8355-bae46de404ad.png)

I opened a 2k php file and did nothing. After 60 seconds I went to open another file and noticed that Komodo had become unresponsive. It crashed a few seconds after I tried interacting with it. 

- I reopened Komodo. 
- Opened another 2k php file. 
- After 60 seconds I opened a small 1k file. 
- I clicked inside both edit panels, mousing around, selecting text, etc. No actual editing.
- Then I left it for a few minutes. 
- After four and  a half minutes total elapsed time I opened three more files (all <4k) within about ten seconds, and Komodo crashed a few seconds later.

**Comment by AvalonNZ** [comment]
FYI - I also downloaded and installed Komodo EDIT 11 since Komodo Edit 7 was working fine and I'm using that as an interim solution. 

Komodo EDIT 11 installed fine and crashed immediately after I opened a file (no dialog box - just a crash to desktop). 

So whatever this is, it's part of Komodo IDE 9, 10, and 11 and Komodo EDIT 11 (which I presume is largely the same codebase) but is NOT part of Komodo EDIT 7. I know that doesn't narrow it down much but you never know what's going to provide the clue you need. 

As this probably won't be solved before you break for Xmas, I'd like to thank you both for your efforts, and wish you all a Happy Holidays. It's about 79F degrees here in New Zealand and we're baking in the summer heat and planning our Xmas beach escapes. We'll be thinking of you guys :-)

**Comment by Naatan** [comment]
hah yeah sorry we've been quiet over the holidays. We've all been enjoying some well deserved days off.

This ticket was just opened and solved with a resolution. The issue ended up being with his machine and software: https://github.com/Komodo/KomodoEdit/issues/3412

I suspect something similar is going on on your end - something about your setup is causing a conflict of some sort. Can you think of anything that might be related?

**Comment by AvalonNZ** [comment]
Hi Naatan,

EUREKA!

I reviewed his log and in fact I DO use the X-Mouse Button Control on my machine. 
I was using version 2.13. 
Started Komodo and opened a small file and then a bigger one. Komodo locked up immediately and then crashed. 

I updated X-Mouse Button Control to version 2.17 and rebooted the PC.
Started Komodo and opened the bigger file. Komodo locked up immediately and then crashed. 

Uninstalled X-Mouse Button Control.

Started Komodo and opened a the bigger file. Then opened several of the biggest files I could fine (~1500 lines).
NO PROBLEMS.

Thanks for your patience guys, and thanks to the guy who found the connection to X-Mouse Button Control.

I'm calling this one solved.
